### PR TITLE
MM-13083 - Removing refresh button from some pages

### PR DESCRIPTION
**Pull request by asaadmahmood**
#### Summary
MM-13083 - Removing refresh button from some pages
Also updated the pulsing animation for the loading bars.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13083

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
IOS Simulator, iPhoneX, 12.2
Android Studio, Pixel, 7.11

#### Screenshots
Updates

### CODE REVIEW

**Comment by enahum** [comment]
@asaadmahmood build failures
```
/home/ubuntu/jenkins/workspace/mm/mattermost-mobile-pr/app/screens/channel/channel_post_list/channel_post_list.js
  18:8  error  'PostListRetry' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by asaadmahmood** [comment]
There is also one more thing that's missing, which is a back or a logout button on the Team not found screen. There is no simple way for a user to navigate back on IOS I think, but since that was not part of this ticket, we can fix that in a separate one.

**Comment by asaadmahmood** [comment]
@migbot Not sure if we have a ticket on that.
We can just add the logout bar at the top. 
Design attached:

image

**Comment by enahum** [comment]
@asaadmahmood I believe adding the bar here will be a bit cumbersome, maybe what we can do is add an additional action below `Try again`

File: app/screens/channel/channel_base.js [hunk 1]
```diff
@@ -255,13 +257,24 @@ export default class ChannelBase extends PureComponent {
             theme,
         } = this.props;
 
+        const errorTitle = {
```
Comment (enahum) [inline app/screens/channel/channel_base.js]: can we move this variables inside the `if (channelRequestFailed)` statement? no need to define them if they are not being used.

File: app/screens/channel/channel_base.js [hunk 2]
```diff
@@ -255,13 +257,24 @@ export default class ChannelBase extends PureComponent {
...
+            defaultMessage: 'Make sure you have an active connection and try again.',
+        };
+
         if (!currentChannelId) {
             if (channelsRequestFailed) {
-                const PostListRetry = require('app/components/post_list_retry').default;
                 return (
-                    <PostListRetry
-                        retry={this.retryLoadChannels}
+                    <FailedNetworkAction
```
Comment (enahum) [inline app/screens/channel/channel_base.js]: we need to Lazy load this with an inline require as the previous `PostListRetry` that we had

File: app/screens/channel/channel_post_list/channel_post_list.js [hunk 3]
```diff
@@ -181,18 +180,15 @@ export default class ChannelPostList extends PureComponent {
...
         let component;
 
         if (visiblePostIds.length === 0 && channelRefreshingFailed) {
+            const Loading = require('app/components/channel_loader').default;
             component = (
-                <PostListRetry
-                    retry={this.loadPostsRetry}
-                    theme={theme}
-                />
+                <Loading channelIsLoading={true}/>
```
Comment (enahum) [inline app/screens/channel/channel_post_list/channel_post_list.js]: Why was the PostListRetry removed from here? in case this fails it will show the channel loader forever instead of providing the user with more feedback

Comment (asaadmahmood) [reply app/screens/channel/channel_post_list/channel_post_list.js]: Because we already have a bar at the top that shows the user the information.
The post list retry is not showing any information, just a refresh button.

File: app/screens/channel/channel_base.js (hunk 1 above)
Comment (asaadmahmood) [reply app/screens/channel/channel_base.js]: Sure.

File: app/screens/channel/channel_post_list/channel_post_list.js (hunk 3 above)
Comment (enahum) [reply app/screens/channel/channel_post_list/channel_post_list.js]: yeah, but I rather not have an endless channel loading, to me that is as bad as the previous one.. Maybe show the cloud image with some text or something other than an endless channel loader

Comment (asaadmahmood) [reply app/screens/channel/channel_post_list/channel_post_list.js]: @enahum We have another larger ticket to look at the different states.
I can ask Andrew to look into the design and the message that needs to be there for it.
https://mattermost.atlassian.net/browse/MM-16251

**Comment by asaadmahmood** [comment]
@enahum Updated the posts error state with this layout based on the proposal.
Wasn't able to get the markdown working so will just submit this, so we can take this in, and then move to that other ticket later.

Screenshot 2019-07-24 at 9 30 31 PM

**Comment by saturninoabril** [comment]
@asaadmahmood For team not found, why does the onPress (or retry) only applied to `try again` and not the entire error description?  Does the user has to specifically click that phrase just to retry?

**Comment by asaadmahmood** [comment]
@saturninoabril Yes, the try again is styled as a link.
See screenshot.

Screenshot 2019-07-26 at 1 45 52 AM

File: app/screens/error_teams_list/error_teams_list.test.js [hunk 4]
```diff
@@ -32,32 +26,31 @@ describe('ErrorTeamsList', () => {
...
+        expect(baseProps.actions.loadMe).toHaveBeenCalledTimes(1);
+        expect(baseProps.actions.loadMe).toBeCalledWith();
+
+        expect(baseProps.actions.resetToChannel).toHaveBeenCalledTimes(1);
+        expect(baseProps.actions.resetToChannel).toBeCalledWith({disableTermsModal: true});
+
+        expect(baseProps.actions.selectDefaultTeam).toHaveBeenCalledTimes(1);
+        expect(baseProps.actions.selectDefaultTeam).toBeCalledWith();
+
+        expect(wrapper.state('loading')).toEqual(false);
```
Comment (saturninoabril) [inline app/screens/error_teams_list/error_teams_list.test.js]: `onPress` cannot be called on this component.  It can be tested in FormattedText component but since it's also passed into react-native's Text component, it don't make sense to test since it's already covered by the library itself.  Added unit test is to check functionality of getUserInfo function.

**Comment by saturninoabril** [comment]
Thanks @asaadmahmood.
Done QA review
- tested on Android and iOS
- added unit test

**Comment by asaadmahmood** [comment]
@esethna Updated the terms of service page to be the same as the other ones.
image

**Comment by enahum** [comment]
@asaadmahmood tests are failing

```
Summary of all failing tests
FAIL app/screens/error_teams_list/error_teams_list.test.js
  ● ErrorTeamsList › should call for userInfo on retry

    TypeError: wrapper.find(...).props(...).onRetry is not a function

      57 |         );
      58 | 
    > 59 |         wrapper.find(FailedNetworkAction).props().onRetry();
         |                                                   ^
      60 |         await loadMe();
      61 |         expect(selectDefaultTeam).toHaveBeenCalledTimes(1);
      62 |     });

      at onRetry (app/screens/error_teams_list/error_teams_list.test.js:59:51)
```

File: app/components/failed_network_action/index.js [hunk 5]
```diff
@@ -25,13 +24,27 @@ export default class FailedNetworkAction extends PureComponent {
             defaultMessage: 'No internet connection',
         },
         errorDescription: {
-            id: t('mobile.failed_network_action.description'),
-            defaultMessage: 'There seems to be a problem with your internet connection. Make sure you have an active connection and try again.',
+            id: t('mobile.failed_network_action.shortDescription'),
+            defaultMessage: '{type} will load when you have an internet connection or {refresh}.',
```
Comment (enahum) [inline app/components/failed_network_action/index.js]: a bit concern on translations here, at lease in spanish I would use an article before the word "messages" but is not really clear that I would need one here. I get that you need the `Try again` action, maybe think about passing the action as a prop instead of the whole `<FormattedText>` element or some other way

File: app/screens/channel/channel_base.js [hunk 6]
```diff
@@ -255,14 +256,47 @@ export default class ChannelBase extends PureComponent {
...
+                                id='mobile.failed_network_action.description.messages'
+                                defaultMessage='messages'
+                            />
+                        ),
+                        refresh: (
+                            <FormattedText
+                                id='mobile.failed_network_action.retry'
+                                defaultMessage='try again'
+                                style={style.link}
+                                onPress={this.retryLoadChannels}
```
Comment (enahum) [inline app/screens/channel/channel_base.js]: if you end up passing this action as a prop of `<FailedNetworkAction>` then you don't need the changes to the styles here, then you could revert the changes in `channel.ios.js` and `channel.android.js`

File: app/screens/channel/channel_post_list/channel_post_list.js [hunk 7]
```diff
@@ -185,13 +186,38 @@ export default class ChannelPostList extends PureComponent {
...
+                        <FormattedText
+                            id='mobile.failed_network_action.description.messages'
+                            defaultMessage='messages'
+                        />
+                    ),
+                    refresh: (
+                        <FormattedText
+                            id='mobile.failed_network_action.retry'
+                            defaultMessage='try again'
+                            style={style.link}
```
Comment (enahum) [inline app/screens/channel/channel_post_list/channel_post_list.js]: same here

File: app/screens/error_teams_list/error_teams_list.js [hunk 8]
```diff
@@ -86,31 +77,53 @@ export default class ErrorTeamsList extends PureComponent {
...
+                        id='mobile.failed_network_action.description.teams'
+                        defaultMessage='Teams'
+                    />
+                ),
+                refresh: (
+                    <FormattedText
+                        id='mobile.failed_network_action.retry'
+                        defaultMessage='try again'
+                        style={style.link}
+                        onPress={this.getUserInfo}
```
Comment (enahum) [inline app/screens/error_teams_list/error_teams_list.js]: and here

File: app/screens/flagged_posts/flagged_posts.js [hunk 9]
```diff
@@ -197,13 +197,36 @@ export default class FlaggedPosts extends PureComponent {
...
+                        id='mobile.failed_network_action.description.messages'
+                        defaultMessage='messages'
+                    />
+                ),
+                refresh: (
+                    <FormattedText
+                        id='mobile.failed_network_action.retry'
+                        defaultMessage='try again'
+                        style={style.link}
+                        onPress={this.retry}
```
Comment (enahum) [inline app/screens/flagged_posts/flagged_posts.js]: :P

File: app/screens/pinned_posts/pinned_posts.js [hunk 10]
```diff
@@ -194,13 +194,36 @@ export default class PinnedPosts extends PureComponent {
...
+                        id='mobile.failed_network_action.description.messages'
+                        defaultMessage='messages'
+                    />
+                ),
+                refresh: (
+                    <FormattedText
+                        id='mobile.failed_network_action.retry'
+                        defaultMessage='try again'
+                        style={style.link}
+                        onPress={this.retry}
```
Comment (enahum) [inline app/screens/pinned_posts/pinned_posts.js]: again

File: app/components/failed_network_action/index.js [hunk 11]
```diff
@@ -15,25 +15,32 @@ export default class FailedNetworkAction extends PureComponent {
...
         const style = getStyleFromTheme(theme);
 
+        const errorTitle = {
+            id: t('mobile.failed_network_action.title'),
+            defaultMessage: 'No internet connection',
+        };
+
+        const errorDescription = {
+            id: t('mobile.failed_network_action.shortDescription'),
+            defaultMessage: 'Messages will load when you have an internet connection or {refresh}.',
```
Comment (enahum) [inline app/components/failed_network_action/index.js]: maybe the id and defaultMessage can be a prop too? that way we can change the message depending on the screen you are seeing?

I'm thinking on this props.
action: PropTypes.func -> this is the action that can be taken
actionId: PropTypes.string -> the translation id for the action
actionDefaultMessage: PropTypes.string -> the default message used in the translation for the action
errorId: PropTypes.string -> the id for the translation
errorDefaultMessage: PropTypes.string -> the default message used in the translation for the message.

Comment (asaadmahmood) [reply app/components/failed_network_action/index.js]: Done @enahum

File: app/components/failed_network_action/index.js [hunk 12]
```diff
@@ -13,27 +13,45 @@ import Cloud from './cloud';
...
     static defaultProps = {
-        errorTitle: {
-            id: t('mobile.failed_network_action.title'),
-            defaultMessage: 'No internet connection',
-        },
-        errorDescription: {
-            id: t('mobile.failed_network_action.description'),
-            defaultMessage: 'There seems to be a problem with your internet connection. Make sure you have an active connection and try again.',
-        },
+        actionId: 'mobile.failed_network_action.retry',
```
Comment (enahum) [inline app/components/failed_network_action/index.js]:
```suggestion
        actionId: t('mobile.failed_network_action.retry'),
```

File: app/components/failed_network_action/index.js [hunk 13]
```diff
@@ -13,27 +13,45 @@ import Cloud from './cloud';
...
+        actionDefaultMessage: 'try again',
+        errorId: 'mobile.failed_network_action.shortDescription',
```
Comment (enahum) [inline app/components/failed_network_action/index.js]:
```suggestion
        errorId: t('mobile.failed_network_action.shortDescription'),
```

File: app/components/failed_network_action/index.js [hunk 14]
```diff
@@ -13,27 +13,45 @@ import Cloud from './cloud';
...
-        const {theme, onRetry} = this.props;
+        const {action, actionId, actionDefaultMessage, errorId, errorDefaultMessage, theme} = this.props;
         const style = getStyleFromTheme(theme);
 
+        const errorTitle = {
+            id: t('mobile.failed_network_action.title'),
+            defaultMessage: 'No internet connection',
+        };
+
+        const errorDescription = {
```
Comment (enahum) [inline app/components/failed_network_action/index.js]: you no longer need this variable here

File: app/components/failed_network_action/index.js [hunk 15]
```diff
@@ -42,27 +60,16 @@ export default class FailedNetworkAction extends PureComponent {
...
-                    id={this.props.errorTitle.id}
-                    defaultMessage={this.props.errorTitle.defaultMessage}
+                    id={errorTitle.id}
+                    defaultMessage={errorTitle.defaultMessage}
                     style={style.title}
                 />
                 <FormattedText
-                    id={this.props.errorDescription.id}
-                    defaultMessage={this.props.errorDescription.message}
+                    id={errorDescription.id}
```
Comment (enahum) [inline app/components/failed_network_action/index.js]:
```suggestion
                    id={errorId}
```

File: app/components/failed_network_action/index.js [hunk 16]
```diff
@@ -42,27 +60,16 @@ export default class FailedNetworkAction extends PureComponent {
...
+                    defaultMessage={errorDescription.message}
```
Comment (enahum) [inline app/components/failed_network_action/index.js]:
```suggestion
                    defaultMessage={errorDefaultMessage}
```

File: app/components/failed_network_action/index.js [hunk 17]
```diff
@@ -42,27 +60,16 @@ export default class FailedNetworkAction extends PureComponent {
...
                     style={style.description}
+                    values={errorDescription.values}
```
Comment (enahum) [inline app/components/failed_network_action/index.js]:
```suggestion
                    values={{
                        refresh: (
                            <FormattedText
                                id={actionId}
                                defaultMessage={actionDefaultMessage}
                                style={style.link}
                                onPress={action}
                            />
                        ),
                    }}
```

**Comment by asaadmahmood** [comment]
@enahum Done.

**Comment by enahum** [comment]
@asaadmahmood build failed :S

File: app/components/failed_network_action/index.js (hunk 14 above)
Comment (enahum) [reply app/components/failed_network_action/index.js]: did you forget to remove this?

**Comment by asaadmahmood** [comment]
I did remove it but forgot to save the file 🤦🏻‍♂️
